### DISCUSSION TITLE
Replacing Antlers with Blade

### DISCUSSION THREAD

**Discussion by BrandonOldenhof** [studio1902/statamic-peak#234] [Ideas]
Hello!

With Statamic 3.3 Statamic now has much better support for Laravel Blade.
Are you interested in providing Blade alternatives to the Antlers layouts, or switching to Blade completely in the (near) future?

**Comment by robdekort** [comment]
Hey Brandon. No plans right now as I much prefer using Antlers (Runtime). Sorry!

**Reply by ramoneg** [reply]
Hey Rob - May I ask why you prefer Antlers? And do you use a formatter?

Honestly I just can not start to like it and I really miss the well structured blade approach which just makes more sense to me 😝

**Reply by sjclark** [reply]
If Blade makes more sense to you then use Blade? Everyone has different preferences. I believe Antlers was invented before Laravel Blade, so has been around a very long time.

**Reply by ramoneg** [reply]
Sure, Statamic was there before. But Rob decided at the start of Peak that he will go with Antlers in his amazing starter kit so I just thought I'd ask him.

I was rewriting his package to a blade approach, but I got this gut feeling that I should stick with his one since I can't fully use all the features Antlers provides and I have a feeling that this will be bad in the future. Is it just me or is the Syntax sometimes confusing as hell and has some limits? And I just couldn't get any good auto-formatting with this thing. Kinda annoying.

With Blade I feel way more free and since Statamic is a big name in the Laravel world I hoped Blade was the default by now.

**Reply by sjclark** [reply]
As before, everyone has different preferences. The same argument could be made to use liquid, handlebars, twig, or even just raw php etc.

Rob decided at the start of building a starter kit for Statamic to build it in the language that Statamic is designed for - it would have had a tiny fraction of usage or contributions if it had been made in Blade.

I think you're confusing your *preference* for blade, with Antlers being bad. Conversely, theres lots of people who prefer Antlers and dislike blade. Hence why you're welcome to use whatever you want - my understanding was the [majority of functionality is still possible](https://statamic.dev/blade) within Blade - you might find the docs aren't quite as good - but thats because the whole CMS is designed around Antlers - and if you're wanting to use some different external language then you'll need to adapt slightly.

Theres loads of docs on Antlers, [theres a book](https://stillat.com/books/antlers-building-beautiful-websites-with-statamic), theres [IDE/editor plugins](https://antlers.dev/docs/formatting). You can even use antlers [**within** blade](https://github.com/statamic/cms/pull/8692), can use raw php etc. Theres places to report issues if you find the formatting doesn't work well.

While you might think Statamic is a big name it still has a tiny team, with a miniscule fraction of the resources of Laravel - would be a big undertaking to develop complete codebase and docs support for two languages in parallel. But if you think it's worthwhile that all of Statamic start supporting blade as a first-class language (with maybe a language toggle in the docs) then it's probably worth posting as an idea in the Statamic Ideas Repo. Same goes if there are [specific features](https://github.com/statamic/cms/pull/2128) you think should be added for blade.

In any case, I'm sure @robdekort will weigh in when he wakes up 👍

**Reply by robdekort** [reply]
Yep, what Sam says.

I don't use a formatter. However the VS Code plugin has built in formatting you can turn on. Personally I don't like it when software dictates how my code should look. And depending on the complexity of my templates, I tend to format stuff differently. You can probably see that in some of the Peak templates.

I like Antlers, because the CMS is built around it. I'd rather embrace that as opposed to bending the software to a template language that isn't fully supported, and has one leg into pure PHP. Which in my opinion shouldn't be the case in a templating language. 

This wouldn't be my preferred way of templating:
```blade
@foreach($tag = Statamic::tag('collection:blog')->fetch())
    <li>{{ $title }}</li>
@endforeach
```

I'd much prefer:
```antlers
{{ collection:blog  }}
   {{ title }}
{{ /collection:blog }}
```

I understand this taste and preference. And neither of those has to be defended. Pick whatever you like.

For me, Antlers has countless tags and modifiers built around all of the features Statamic has to offer. All of them are documented and suit my needs. With partials and parameters I can work as dry and structured as I want. The suggestion that you can't do that with Antlers, gives me the impression you haven't given it a serious shot. Or perhaps you were put off by the syntax, because it's not what you'd want it to be? [This doc](https://statamic.dev/antlers#overview) is basically all you need to know.

I can do [advanced stuff](https://github.com/studio1902/statamic-peak-tools/blob/main/resources/views/components/_picture.antlers.html) in Antlers. The runtime parser is super fast and mostly bug free. Whenever there's a bug it often gets fixed within hours. Literally. Thanks to @JohnathonKoster for that. There's great syntax highlighting (and more) for VS Code. It works with Livewire. It works with Torchlight. There's debugging tools and a profiler available. It's a full suite. 

However, it's not Blade, and neither is Peak. Can't help you with that. 🤷🏻‍♂️

**Reply by ramoneg** [reply]
Thank you for your reply!

**Reply by robdekort** [reply]
You're welcome. What I think is very important for folks to understand is that Statamic is a CMS built on top of Laravel. It's not necessarily a CMS for Laravel developers. I can't see it being marketed that way either. A lot of folks in the community start with Statamic without ever having done something with Laravel or Blade. For a lot of them Antlers is a much friendlier welcome than Blade would be. 

One last thing I'd like to add from a personal perspective. To me the [discussion around](https://github.com/statamic/cms/discussions/6189) Antlers - _usually from Laravel devs_ - can come across as a bit entitled. My wish is for everyone who feels this way about Antlers in Statamic is to let get go of the desire for the CMS to be tailored to them specifically. Both template languages can happily exist together.
